### DISCUSSION TITLE
Add jianpu support

### DISCUSSION THREAD

**Discussion by jhuang97** [musescore/MuseScore#22698] [Ideas]
Jianpu is a musical notation system using numbers, which is popular in China and elsewhere.  I have previously used some web app that can edit and display jianpu along with song lyrics; it had its own special markup language.  For me, it would be nice to be able to edit jianpu scores in Musescore and switch a staff between jianpu and staff notation.

There have already been multiple attempts at implementing jianpu in Musescore:

- Efforts in 2017-2018. Added a new StaffGroup and StaffType for jianpu, and created several new classes for elements of jianpu notation, including Jianpu notes and chords. However, these changes were on top of a much earlier version of Musescore. #2957, #3614 
- More recent attempt by Auase, getting as far as rendering single-line Jianpu. First pull request: #19395. Second pull request with much discussion: #19421.

**Comment by cbjeukendrup** [comment]
The current status about Jianpu is that Auase had created those PRs you mention, but those could not directly be merged because they needed to be improved (at least code-technically). Auase is planning to work on these PRs again in the future, but is currently busy with other things, and not yet sure when he/she has time to come back to it.

**Reply by TaoxiaoyuZN** [reply]
We will wait patiently and hope it can be realized as soon as possible

**Comment by King-of-Infinite-Space** [comment]
In case anyone wants a hacky workaround, here's a plugin [MuseScore-Integer-Notation](https://github.com/Mind-Ext/MuseScore-Integer-Notation)

**Reply by jhuang97** [reply]
That looks pretty cool, even though it's not exactly the same as jianpu

**Reply by jhuang97** [reply]
When you re-position the notes vertically, you align the top notes, not the bottom notes.  Is the idea that the top notes are more likely to be the melody notes?

**Reply by King-of-Infinite-Space** [reply]
Honestly I didn't think much about it, but your interpretation sounds reasonable. Could make bottom align an option next time.

**Comment by SubjunctiveQuaver** [comment]
Not sure if there's been any progress on this, but Muse VP of Product Tantacrul released a video 'Notation Must Die: The Battle For How We Read Music' on his YouTube channel in November 2023. It has a brief clip showcase of jianpu support in Musescore.

At that time, it was 'in development', but I'd love to know where it's up to now 1.5 years later. It seems the latest version doesn't have the functionality enabled yet.

[See here (1:14:30) for the MuseScore Jianpu showcase](https://youtu.be/Eq3bUFgEcb4?si=q7B9HeCNi9pPP6t7&t=4470)

EDIT: after doing some more searching, it was likely linked to the closed PR https://github.com/musescore/MuseScore/pull/19421, so progress seems to have stalled.

**Comment by BLumia** [comment]
Also adding a link here:

- https://github.com/musescore/MuseScore/issues/25919

Just in case people might want to subscribe to that issue to know if there are related updates.

**Comment by odhot** [comment]
i made this .. https://www.youtube.com/watch?v=COU9Oo-0gBA
